### PR TITLE
Support : Leverage Accelerate for object detection/segmentation models 

### PR DESCRIPTION
**Pull request by Tanmaypatil123**
# What does this PR do?

Adding support for multi-GPU training in 6 object detection models and segmentation models.

Fixes # #28309

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@NielsRogge

### CODE REVIEW

**Comment by NielsRogge** [comment]
Hi, thanks for your PR. It looks like there's an issue with the segmentation models.

**Comment by Tanmaypatil123** [comment]
@NielsRogge could you take a look at this PR again . I have made required changes for segmentation models.

File: src/transformers/models/conditional_detr/modeling_conditional_detr.py
Comment (amyeroberts) [inline src/transformers/models/conditional_detr/modeling_conditional_detr.py:24]: This needs to be behind a protected import: 

```py
if is_accelerate_available():
    from accelerate import PartialState
    from accelerate.utils import reduce
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on this @Tanmaypatil123! 

Just a comment on the imports

**Review by amyeroberts — APPROVED** [review]
Thanks for iterating! Changes look good to me. 

Before merging could you: 
* Run the slow tests for these models to confirm everything still passes? e.g. `RUN_SLOW=1 pytest tests/models/conditional_detr/`
* Test at least one of these models in a single and multi-GPU environment with the loss calculation i.e. labels passed in? 
* Resolve conflicts

**Comment by Tanmaypatil123** [comment]
hey @amyeroberts removed conflicts .but whenever I tried to run test cases by `RUN_SLOW=1 pytest tests/models/conditional_detr/`  I am getting following error : 
image

**Comment by amyeroberts** [comment]
@Tanmaypatil123 - Not sure exactly what's causing this. The function `is_g2p_en_available` was added in #23439. However, we haven't seen this import error in our CI runs which validate slow tests and I'm able to run `RUN_SLOW=1 pytest tests/models/conditional_detr/` locally on `main`.

In a python session, are you about to run: 
```
from transformers.utils import is_g2p_en_available
```
? 

Could you try rebasing on main to make sure all updates/fixes are included in this branch?

**Comment by ArthurZucker** [comment]
The issue is related to evaluate I think: https://github.com/huggingface/peft/issues/1351

**Comment by ArthurZucker** [comment]
cc @Tanmaypatil123 let's rebase this and I think we could merge this!

**Comment by Tanmaypatil123** [comment]
@ArthurZucker Changes are approved but there is issue with segmentation model; test cases fails for it and i could not figure out what the problem is.

**Comment by NielsRogge** [comment]
@Tanmaypatil123 you just need to fix the upstream conflicts, like so:
```
git remote add upstream https://github.com/huggingface/transformers.git
git fetch upstream
git merge upstream/main
```

**Comment by Tanmaypatil123** [comment]
@ArthurZucker I have resolved the conflicts

**Comment by amyeroberts** [comment]
@Tanmaypatil123 Great! 

For the failing tests, there's two which look like they might be related to this PR which will needed to be addressed before merge: 

```
FAILED tests/models/yolos/test_modeling_yolos.py::YolosModelTest::test_pipeline_image_feature_extraction - TypeError: forward() got an unexpected keyword argument 'pixel_mask'
FAILED tests/models/yolos/test_modeling_yolos.py::YolosModelTest::test_pipeline_object_detection - TypeError: forward() got an unexpected keyword argument 'pixel_mask'
```

For the build documentation tests, there was a recent fix merged into main. Rebasing to include this should resolve.

**Comment by Tanmaypatil123** [comment]
> @Tanmaypatil123 Great!
> 
> For the failing tests, there's two which look like they might be related to this PR which will needed to be addressed before merge:
> 
> ```
> FAILED tests/models/yolos/test_modeling_yolos.py::YolosModelTest::test_pipeline_image_feature_extraction - TypeError: forward() got an unexpected keyword argument 'pixel_mask'
> FAILED tests/models/yolos/test_modeling_yolos.py::YolosModelTest::test_pipeline_object_detection - TypeError: forward() got an unexpected keyword argument 'pixel_mask'
> ```
> 
> For the build documentation tests, there was a recent fix merged into main. Rebasing to include this should resolve.

Don't know how that test case is failing. I didn't make any changes to the parameters that are provided to `YolosForObjectDetection`. should we add parameters or is there any change in the test case?

**Comment by amyeroberts** [comment]
@Tanmaypatil123  Indeed. It's quite weird these tests are failing as they don't seem related to this PR. They aren't however failing on any other PRs. What I think is happening is that some tests weren't collected by the test fetched (cc @ydshieh for reference) when the image feature extraction pipeline was added. And now, because this PR touches yolos they're being run now. I'm going to look into it and let you know asap.

**Comment by Tanmaypatil123** [comment]
@amyeroberts Done 😀 Thanks for the help. Huggingface has the best open source maintainers.

**Comment by amyeroberts** [comment]
@Tanmaypatil123 Thanks for this great contribution and patience with our failing CI tests!

**Pull request merged**
